forgerock-authenticator/src/main/java/org/forgerock/android/auth/OathMechanism.java

```diff
@@ -200,8 +201,15 @@ public T setTimeCreated(Calendar timeCreated) {
         /**
          * Produce the described OathMechanism Token.
          * @return The built Token
+         * @throws MechanismCreationException If an issuer or accountName were not provided.
          */
-        protected OathMechanism build() {
+        protected OathMechanism build() throws MechanismCreationException {
+            if(issuer == null || issuer.isEmpty()) {
```

Comment:
I think it is worth moving org.forgerock.android.auth.StringUtils from auth to core, and use org.forgerock.android.auth.StringUtils#isEmpty

Reply:
I think it's a good idea. But I would recommend handle it in a separated PR.